### PR TITLE
Feat: Text-Equals

### PR DESCRIPTION
**Pull request by degrootsam**

### CODE REVIEW

File: functions/equals-text/1.0/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        case 'endsWith':
+            return { result: leftValue.endsWith(rightValue) };
+        case 'regMatch':
+            const regex = new RegExp(rightValue);
+            return { result: regex.test(leftValue) };
+        default: 
+            return { result: false }
+        }
+    }
+    export default equalsText;
```
Comment (justin-betty) [inline functions/equals-text/1.0/index.js:27]: Please add a empty line at the end

File: test/compare-values.test.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+  assert(equal, { result: false });
+  assert(notEqual, { result: true });
+  assert(contains, { result: true });
+  assert(notContains, { result: false });
+  assert(toLower, { result: true });
+  assert(startsWith, { result: true });
+  assert(endsWith, { result: true });
+  assert(regMatch, { result: true });
+});
```
Comment (justin-betty) [inline test/compare-values.test.js:20]: Please add a empty line here

File: functions/equals-text/1.0/index.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const equalsText = async ({ leftValue, rightValue, comparator, toLower }) => {
+
+    let res = false;
+    if (toLower) {
+        leftValue = leftValue.toLowerCase();
+        rightValue = rightValue.toLowerCase();
+    }
+
+    switch (comparator) {
```
[the remark this answers is not in the document]
Comment (degrootsam) [reply functions/equals-text/1.0/index.js:9]: Check earlier comments made by Thomas

File: functions/equals-text/1.0/index.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+const equalsText = async ({ leftValue, rightValue, comparator, toLower }) => {
+
+    let res = false;
```
Comment (justin-betty) [inline functions/equals-text/1.0/index.js:3]: Can you do `let result;`

File: functions/equals-text/1.0/index.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        case 'endsWith':
+            res = leftValue.endsWith(rightValue);
+        case 'regMatch':
+            const regex = new RegExp(rightValue);
+            res = regex.test(leftValue);
+        default: 
+            res = false;
+        }
+
+        return {
```
Comment (justin-betty) [inline functions/equals-text/1.0/index.js:29]: Here you can do 

`return { result }`

File: functions/equals-text/1.0/index.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            result: res
+        }
+
+}
+export default equalsText;
```
Comment (justin-betty) [inline functions/equals-text/1.0/index.js:34]: Add empty line at the end

**Comment by justin-betty** [comment]
Can you also add a unit test

File: functions/equals-text/1.0/index.js (hunk 6 above)
Comment (jwielinga) [reply functions/equals-text/1.0/index.js:34]: Better if you add it to prettier

File: test/text-equals.test.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
+test('Using "eq" returns true if the left value equals the right value ', async () => {
+  const result = await $app['equalsText 1.0']({ leftValue: 'Lorum Ipsum', rightValue: 'Lorum Ipsum', comparator: 'eq', toLower: false });
```
Comment (jwielinga) [inline test/text-equals.test.js:2]: Don't declare the same parameter everytime. Declare shared variable if values are reused

**Review by jwielinga — CHANGES_REQUESTED** [review]

**Comment by degrootsam** [comment]
Due to fork, closed pull request. See: https://github.com/bettyblocks/block-store-action-functions/pull/83

**Pull request closed without merging**